src/popup/container.js

```diff
@@ -201,6 +231,27 @@ class PopupContainer extends Component {
         return this.props.pauseValues.map(pauseValueToOption)
     }
 
+    handleAddBookmark = () => {
+        if (
+            this.state.bookmarkBtn === constants.BOOKMARK_BTN_STATE.UNBOOKMARK
+        ) {
+            this.createBookmarkByExtension(this.state.url)
+        } else if (
+            this.state.bookmarkBtn === constants.BOOKMARK_BTN_STATE.BOOKMARK
+        ) {
+            this.removeBookmarkByUrl(this.state.url)
+        }
+        this.setState(state => ({
```

Comment:
No reason to update the local state here if you call `window.close` straight after (since the state won't exist anymore).